src/extractor/guidance/turn_analysis.cpp

```diff
@@ -123,7 +126,9 @@ TurnAnalysis::transformIntersectionIntoTurns(const Intersection &intersection) c
 
 Intersection TurnAnalysis::getIntersection(const NodeID from_nid, const EdgeID via_eid) const
 {
-    return intersection_generator(from_nid, via_eid);
+    auto intersection = intersection_generator(from_nid, via_eid);
+    const auto node_at_intersection = node_based_graph.GetTarget(via_eid);
+    return intersection_optimizer.Process(node_at_intersection, std::move(intersection));
```

Comment:
This is what I meant above: operator() for the generator vs Process for the optimizer